packages/google_maps_flutter/google_maps_flutter_web/lib/src/dom_window_extension.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:js_interop';
+
+import 'package:web/web.dart';
+
+/// This extension type exists to handle unsupported features by certain browsers.
+@JS()
+extension type WindowWithTrustedTypes(Window _) implements JSObject {
+  /// Get the `trustedTypes` object from the window, if it is supported.
+  @JS('trustedTypes')
+  external TrustedTypePolicyFactory? get trustedTypesNullable;
+}
```

Comment:
Another way of doing this:

* https://github.com/flutter/packages/blob/91d11d696196880fe3cd4817c08362e59d7b5c56/packages/google_identity_services_web/lib/src/js_interop/package_web_tweaks.dart#L11-L30

That way when using window from `package:web`, you get the nullable value directly, without having to wrap it in `WindowWithTrustedTypes(window)` to check.

There's also a `has` property in `dart:js_interop_unsafe` that can be used to do: `window.hasProperty('trustedTypes')` (but I have never done so). See: https://github.com/dart-lang/web/issues/181

Reply:
Good catch. I kept only the nullable version, since the code already handles the existence (or not) of trusted types.